OP2Maps/MapData.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include "MapHeader.h"
+#include "TileData.h"
+#include "TileSetSource.h"
+#include "TileInfo.h"
+#include "TerrainType.h"
+#include "ClipRect.h"
+#include "../StreamReader.h"
```

Comment:
Consider using a forward declare:
class FileStreamReader;
class SeekableStreamReader;
class StreamReader;

We don't actually need the full class definition, since we are only declaring a pointer to a class of this type. The compiler only needs to know the size of a pointer, which doesn't depend on the specifics of the class.

Using a forward declare instead avoids having to read an parse an extra header file (along with anything else that recursively includes), which can improve compile times, and reduce the amount of code that is rebuilt if the header file is edited.

Reply:
I'm trying to avoid using the word map by itself here to keep away from using the same name as the standard library map container. Hence the class MapData and the folder OP2Maps. I'm looking at wrapping everything in the OP2Maps folder in an OP2Maps namespace. 

I'm certainly open to other naming conventions here if other things make more sense. Probably best to choose a good name now before Leeor starts integrating it though.

Good idea about the forward declare of the StreamReader class. I'll look at adding this soon.

Reply:
When attempting to forward delcare the three StreamReader classes, I am getting a lot of the following error:

C2027 use of undefined type ‘StreamReader’

For example: Stream Reader here is throwing the error:

void MapData::ReadMapHeader(StreamReader& streamReader)
{
	streamReader.Read((char*)&mapHeader, sizeof(mapHeader));
}

Either I'm missing something about Forward Declares or this isn't a valid use of them???

Reply:
If that is coming from the .cpp file, then add the `#include` there. You will very likely need a full include from the .cpp file.